rabit/include/rabit/internal/utils.h

```diff
@@ -180,6 +162,19 @@ inline std::FILE *FopenCheck(const char *fname, const char *flag) {
   return fp;
 }
 }  // namespace utils
+
+// Can not use std::min on Windows with msvc due to:
```

Comment:
We could define `NOMINMAX` macro to disable `MIN` and `MAX` macros from Windows.h.

Reply:
That's a bit weird.  Can we just stick with current approach?  That feels like an invitation to header include ordering issues.

Reply:
The current approach is fine.